services-custom/s3-transfer-manager/src/test/java/software/amazon/awssdk/transfer/s3/samples/S3TransferManagerSamples.java

```diff
@@ -0,0 +1,222 @@
+import software.amazon.awssdk.transfer.s3.model.UploadDirectoryRequest;
+import software.amazon.awssdk.transfer.s3.progress.LoggingTransferListener;
+
+/**
+ * Contains code snippets that will be used in the Javadocs of {@link S3TransferManager}
+ */
+public class S3TransferManagerSamples {
+
+    public void defaultClient() {
+        // @start region=defaultTM
+        S3TransferManager transferManager = S3TransferManager.create();
+        // @end region=defaultTM
+    }
+
+    public void customClient() {
+        // @start region=customTM
+        S3AsyncClient s3AsyncClient = S3AsyncClient.crtBuilder()
+                                                   .credentialsProvider(DefaultCredentialsProvider.create())
+                                                   .region(Region.US_WEST_2)
+                                                   .targetThroughputInGbps(20.0)
+                                                   .minimumPartSizeInBytes(8 * MB)
+                                                   .build();
+
+        S3TransferManager transferManager =
+            S3TransferManager.builder()
+                             .s3AsyncClient(s3AsyncClient)
+                             .build();
+        // @end region=customTM
+    }
+
+    public void downloadFile() {
+        // @start region=downloadFile
+        S3TransferManager transferManager = S3TransferManager.create();
+
+        FileDownload download =
+            transferManager.downloadFile(d -> d.getObjectRequest(g -> g.bucket("bucket").key("key")).destination(Paths.get(
+                "myFile.txt")));
+
+        // Wait for the transfer to complete
+        download.completionFuture().join();
+        // @end region=downloadFile
+    }
+
+    public void download() {
+        // @start region=download
+        S3TransferManager transferManager = S3TransferManager.create();
+
+        // Initiate the transfer
+        Download<ResponseBytes<GetObjectResponse>> download =
+            transferManager.download(d -> d.getObjectRequest(g -> g.bucket("bucket")
+                                                                   .key("key"))
+                                           .responseTransformer(AsyncResponseTransformer.toBytes()));
+        // Wait for the transfer to complete
+        download.completionFuture().join();
+        // @end region=download
+    }
+
+    public void resumeDownloadFile() {
+        // @start region=resumeDownloadFile
+        S3TransferManager transferManager = S3TransferManager.create();
+
+        // Initiate the transfer
+        FileDownload download =
+            transferManager.downloadFile(d -> d.getObjectRequest(g -> g.bucket("bucket").key("key")).destination(Paths.get(
+                "myFile.txt")));
+
+        // Pause the download
+        ResumableFileDownload resumableFileDownload = download.pause(); // @link substring="pause" target="software.amazon.awssdk.transfer.s3.model.FileDownload#pause()"
```

Comment:
@link seems cool. Where should we be using it?

Reply:
Yup. We can use it for any method we want to link to its Javadoc. The link is not super visible from the javadoc site though
![Screen Shot 2022-10-06 at 1 27 13 PM](https://user-images.githubusercontent.com/33073555/194411941-f7544fb5-3c55-4fc1-9b17-c131cd1c398b.png)